src/index.js

```diff
@@ -1,33 +1,133 @@
+/* eslint-disable no-loop-func */
+/* eslint-disable no-restricted-globals */
 import './styles/style.css';
 
-const todoList = [
-  {
-    discription: 'wash the dishes',
-    completed: false,
-    index: 0,
-  },
-  {
-    discription: 'complete To Do list project',
-    completed: false,
-    index: 1,
-  },
-  {
-    discription: 'implement project code review suggestions',
-    completed: false,
-    index: 2,
-  },
-];
+class TaskManager {
+  constructor(taskName) {
+    this.taskName = taskName;
+    this.completed = false;
+    this.itemIndex = 0;
+  }
+
+  addTask = (todoList) => {
+    const tasks = todoList;
+    localStorage.setItem('todos', JSON.stringify(tasks));
+    location.reload();
+  };
+
+  // remove task from localStorage && create a todoList
+  removeTask = (task) => {
+    const oldTodoList = JSON.parse(localStorage.getItem('todos'));
+    const currentTodoList = [];
+
+    // get all list todo's not deleted
+    oldTodoList.filter((item) => {
+      if (item.taskName !== task) {
+        currentTodoList.push(item);
+      }
+      return item;
+    });
+    localStorage.setItem('todos', JSON.stringify(currentTodoList));
+    location.reload();
+  };
+
+  // get oldTodoList from localStorage and update appropriate todo task.
+  updateTask = (newText, todoIndex) => {
+    const oldTodoList = JSON.parse(localStorage.getItem('todos'));
+    oldTodoList[todoIndex].taskName = newText;
+    localStorage.setItem('todos', JSON.stringify(oldTodoList));
+    location.reload();
+  };
+}
+
```

Comment:
- [ ] According to the requirements, the functionalities for adding, deleting, and editing tasks should be in a new javascript. Kindly move all these functions to a new file and keeps this `index.js` file neater.